src/apps/api/views/search.py

```diff
@@ -35,75 +51,89 @@ def get(self, request, version="v1"):
+                else:
+                    for object_type in OBJECT_LIST:
+                        matching_types.append({'match': {'_obj_type': object_type}})
+                s = s.query('bool', should=matching_types, minimum_should_match=1)
+                data["results"] = get_results(s)
         return Response(data)
 
-    def _search(self, search, query):
+    def _search(self, search, query, obj_types='ALL'):
         if query and query != ' ':
+            fields = []
+            if 'competition' in obj_types:
+                fields += ["title^3", "description^2", "html_text^2", "created_by"]
+            if 'user' in obj_types:
+                fields += ["email^3", "username^3", "name^3"]
+            if 'profile' in obj_types:
+                fields += ["email^3", "username^3", "name^3"]
+            if obj_types == 'ALL' or 'ALL' in obj_types:
+                fields += ['title^3', 'username^3', 'name^3']
+            # Remove duplicates
+            fields = list(set(fields))
             search = search.query(
                 "multi_match",
                 query=query,
                 type="best_fields",
-                fuzziness=1,
-                fields=["title^5", "description^3", "html_text^2", "created_by"]
+                fuzziness=5,
+                # We cast to a set first to remove any duplicates.
+                fields=fields
             )
-            # s = s.highlight('title', fragment_size=50)
-            # s = s.suggest('suggestions', query, term={'field': 'title'})
         return search
 
-    def _filter(self, search, date_flags, start, end, producer):
-        # This month/this year
-        if date_flags == "this_month":
-            search = search.filter('range', start={
-                'gte': "now/M",
-                'lte': "now/M",
-            })
-        if date_flags == "this_year":
-            search = search.filter('range', start={
-                'gte': "now/y",
-                'lte': "now/y",
-            })
-
-        # Start/end range
-        date_args = {}
-        if start:
-            date_args['gte'] = start
-        if end:
-            date_args['lte'] = end
-        if date_args:
-            date_args['format'] = 'date_optional_time'
-            search = search.filter('range', start=date_args)
-
-        # Active competitions, ones with submissions in the last 30 days
-        if date_flags and date_flags == "active":
-            search = search.filter('term', is_active=True)
-        if producer:
-            search = search.filter('term', producer__id=producer)
+    def _filter(self, search, date_flags, start, end, producer, obj_types='ALL'):
+        if 'competition' in obj_types:
+            # This month/this year
+            if date_flags == "this_month":
+                search = search.filter('range', start={
+                    'gte': "now/M",
+                    'lte': "now/M",
+                })
+            if date_flags == "this_year":
```

Comment:
I think with the way the front end is set up this makes sense as an elif since they're separate conditions, but the back-end might be able to handle both independently currently. I guess this is more of a design decision on whether it makes sense to support both at the same time.